In [ ]:
Data
A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows:
1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29

In [ ]:
a. Build 99% Confidence Interval Using Sample Standard Deviation
Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task.

### 1. Data Extraction and Preparation
First, we need to extract the durability data from the problem description and convert it into a numerical format that Python can process.

In [1]:
import numpy as np
import scipy.stats as stats

# Durability data in millions of characters
data_str = "1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29"
durability_data = np.array([float(x.strip()) for x in data_str.split(',')])

print("Extracted Data:", durability_data)

Extracted Data: [1.13 1.55 1.43 0.92 1.25 1.36 1.32 0.85 1.07 1.48 1.2  1.33 1.18 1.22
 1.29]


### 2. Calculate Sample Statistics
Next, we calculate the sample mean, sample standard deviation, and sample size from our extracted data. These are crucial for constructing the confidence interval.

In [2]:
n = len(durability_data)
sample_mean = np.mean(durability_data)
sample_std = np.std(durability_data, ddof=1) # ddof=1 for sample standard deviation

print(f"Sample Size (n): {n}")
print(f"Sample Mean ($\bar{{x}}$): {sample_mean:.3f} million characters")
print(f"Sample Standard Deviation (s): {sample_std:.3f} million characters")

Sample Size (n): 15
Sample Mean ($ar{x}$): 1.239 million characters
Sample Standard Deviation (s): 0.193 million characters


### 3. Rationale for using the t-distribution
We use the **t-distribution** instead of the normal (z) distribution for constructing the confidence interval for the population mean for the following reasons:

*   **Small Sample Size:** Our sample size is $n = 15$, which is less than 30. When the sample size is small, the t-distribution provides a more conservative (wider) interval, accounting for the increased uncertainty.
*   **Population Standard Deviation Unknown:** We do not know the true population standard deviation ($σ$). Instead, we are using the **sample standard deviation ($s$)** as an estimate. When $σ$ is unknown, the t-distribution is the appropriate choice. If the population standard deviation was known, we would use the z-distribution.

### 4. Determine Critical t-value
To build a 99% confidence interval, we need to find the critical t-value. This value depends on the desired confidence level and the degrees of freedom ($df = n - 1$).

For a 99% confidence interval, the alpha level ($\alpha$) is $1 - 0.99 = 0.01$. Since it's a two-tailed interval, we look for $\alpha/2 = 0.005$ in each tail.

In [3]:
confidence_level = 0.99
alpha = 1 - confidence_level # 0.01
degrees_of_freedom = n - 1

# Get the critical t-value for a two-tailed test
t_critical = stats.t.ppf(1 - alpha/2, degrees_of_freedom)

print(f"Degrees of Freedom: {degrees_of_freedom}")
print(f"Critical t-value for {confidence_level*100}% CI: {t_critical:.3f}")

Degrees of Freedom: 14
Critical t-value for 99.0% CI: 2.977


### 5. Calculate Margin of Error
The margin of error (ME) quantifies the uncertainty in our estimate of the population mean. It is calculated as:

$ME = t_{\alpha/2, df} \times \frac{s}{\sqrt{n}}$

In [6]:
standard_error = sample_std / np.sqrt(n)
margin_of_error = t_critical * standard_error

print(f"Standard Error: {standard_error:.3f}")
print(f"Margin of Error: {margin_of_error:.3f}")

Standard Error: 0.050
Margin of Error: 0.148


### 6. Construct the Confidence Interval
Finally, we construct the 99% confidence interval using the formula:

Confidence Interval = Sample Mean $\pm$ Margin of Error

$CI = \bar{x} \pm ME$

## b. Build 99% Confidence Interval Using Known Population Standard Deviation
Now, let's assume we know the population standard deviation ($\sigma$). In this scenario, we use the **z-distribution** instead of the t-distribution.

### 1. Given Information and Rationale for Z-distribution
*   **Known Population Standard Deviation ($\sigma$):** 0.2 million characters
*   **Confidence Level:** 99% ($\alpha = 0.01$)
*   **Sample Size (n):** 15
*   **Sample Mean ($\bar{x}$):** 1.239 million characters (from previous calculation)

We use the **z-distribution** because the population standard deviation ($\sigma$) is known. When $\sigma$ is known, the sampling distribution of the mean is normal (or approximately normal for large $n$), allowing us to use z-scores to construct the confidence interval.

### 2. Determine Critical Z-value
For a 99% confidence interval, the alpha level ($\alpha$) is $1 - 0.99 = 0.01$. Since it's a two-tailed interval, we look for $\alpha/2 = 0.005$ in each tail.

In [8]:
population_std = 0.2 # million characters
confidence_level = 0.99
alpha = 1 - confidence_level # 0.01

# Get the critical z-value for a two-tailed test
z_critical = stats.norm.ppf(1 - alpha/2)

print(f"Known Population Standard Deviation (\u03C3): {population_std} million characters")
print(f"Critical z-value for {confidence_level*100}% CI: {z_critical:.3f}")

Known Population Standard Deviation (σ): 0.2 million characters
Critical z-value for 99.0% CI: 2.576


### 3. Calculate Margin of Error
The margin of error (ME) for a known population standard deviation is calculated as:

$ME = z_{\alpha/2} \times \frac{\sigma}{\sqrt{n}}$

In [9]:
standard_error_z = population_std / np.sqrt(n)
margin_of_error_z = z_critical * standard_error_z

print(f"Standard Error (with \u03C3 known): {standard_error_z:.3f}")
print(f"Margin of Error (with \u03C3 known): {margin_of_error_z:.3f}")

Standard Error (with σ known): 0.052
Margin of Error (with σ known): 0.133


### 4. Construct the Confidence Interval
Finally, we construct the 99% confidence interval using the formula:

Confidence Interval = Sample Mean $\pm$ Margin of Error

$CI = \bar{x} \pm ME$

In [10]:
confidence_interval_lower_z = sample_mean - margin_of_error_z
confidence_interval_upper_z = sample_mean + margin_of_error_z

print(f"99% Confidence Interval (\u03C3 known): ({confidence_interval_lower_z:.3f}, {confidence_interval_upper_z:.3f}) million characters")
print("\nThis means we are 99% confident that the true mean number of characters printed before a print-head fails lies between ")
print(f"{confidence_interval_lower_z:.3f} million and {confidence_interval_upper_z:.3f} million characters, assuming the population standard deviation is 0.2 million characters.")

99% Confidence Interval (σ known): (1.106, 1.372) million characters

This means we are 99% confident that the true mean number of characters printed before a print-head fails lies between 
1.106 million and 1.372 million characters, assuming the population standard deviation is 0.2 million characters.


In [7]:
confidence_interval_lower = sample_mean - margin_of_error
confidence_interval_upper = sample_mean + margin_of_error

print(f"99% Confidence Interval: ({confidence_interval_lower:.3f}, {confidence_interval_upper:.3f}) million characters")
print("\nThis means we are 99% confident that the true mean number of characters printed before a print-head fails lies between ")
print(f"{confidence_interval_lower:.3f} million and {confidence_interval_upper:.3f} million characters.")

99% Confidence Interval: (1.090, 1.387) million characters

This means we are 99% confident that the true mean number of characters printed before a print-head fails lies between 
1.090 million and 1.387 million characters.


In summary, the two confidence intervals are:

Using Sample Standard Deviation (t-distribution): (1.090 million, 1.387 million characters)
Using Known Population Standard Deviation (z-distribution): (1.106 million, 1.372 million characters)